src/analysis/retail/monk/mistweaver/modules/spells/MistsOfLife.tsx

```diff
@@ -0,0 +1,225 @@
+      this.extraEnvmOverHealing += event.overheal || 0;
+    }
+  }
+
+  handleHeal(event: HealEvent) {
+    const targetId = event.targetID;
+    const spellId = event.ability.guid;
+    if (
+      UNAFFECTED_SPELLS.includes(spellId) ||
+      !this.hotTracker.hots[targetId] ||
+      (!this.hotTracker.hots[targetId][TALENTS_MONK.ENVELOPING_MIST_TALENT.id] &&
+        !this.hotTracker.hots[targetId][SPELLS.ENVELOPING_MIST_TFT.id])
+    ) {
+      return;
+    }
+
+    const hot = this.hotTracker.hots[targetId][TALENTS_MONK.ENVELOPING_MIST_TALENT.id];
+    if (this.hotTracker.fromMistsOfLife(hot)) {
+      this.extraEnvBonusHealing += calculateEffectiveHealing(event, this.envmHealingIncrease);
+    }
+  }
+
+  handleRemApply(event: ApplyBuffEvent | RefreshBuffEvent) {
+    const targetId = event.targetID;
+    const spellId = event.ability.guid;
+    if (
+      UNAFFECTED_SPELLS.includes(spellId) ||
+      !this.hotTracker.hots[targetId] ||
+      !this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hotTracker.fromMistsOfLife(hot)) {
+      this.extraRemApplications += 1;
+    }
+  }
+
+  handleRemHeal(event: HealEvent) {
+    const targetId = event.targetID;
+    const spellId = event.ability.guid;
+    if (
+      UNAFFECTED_SPELLS.includes(spellId) ||
+      !this.hotTracker.hots[targetId] ||
+      !this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    const hot = this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hotTracker.fromMistsOfLife(hot)) {
+      this.extraRemHealing += event.amount || 0;
+      this.extraRemOverHealing += event.overheal || 0;
+      this.extraRemHits += 1;
+    }
+  }
+
+  handleVivifyCast(event: CastEvent) {
+    this.lastVivifyCastTarget = event.targetID || 0;
+    this.countedMainVivifyHit = false;
+  }
+
+  handleVivify(event: HealEvent) {
+    const targetId = event.targetID;
+    if (
+      !this.hotTracker.hots[targetId] ||
+      !this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id]
+    ) {
+      return;
+    }
+    // only count cleave hit on main target
+    if (targetId === this.lastVivifyCastTarget && !this.countedMainVivifyHit) {
+      this.countedMainVivifyHit = true;
+      return;
+    }
+    const hot = this.hotTracker.hots[targetId][SPELLS.RENEWING_MIST_HEAL.id];
+    if (this.hotTracker.fromMistsOfLife(hot)) {
+      this.extraVivCleaves += 1;
+      this.extraVivHealing += event.amount || 0;
+      this.extraVivOverhealing += event.overheal || 0;
+    }
+  }
+
+  statistic() {
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.OPTIONAL(13)}
+        size="flexible"
+        category={STATISTIC_CATEGORY.TALENTS}
+        tooltip={
+          <ul>
+            <li>
+              <SpellLink id={TALENTS_MONK.ENVELOPING_MIST_TALENT.id} /> extra hits:{' '}
```

Comment:
I'm not sure I see the value in the 'extra hits' line items here, since these hots cannot be extended by RM, the extra healing from each hot should be sufficient information